infrastructure/grpc/server_test.go

```diff
@@ -0,0 +1,495 @@
+					gomock.Any(),
+				).Return(
+					&models.MicroVM{
+						ID:      *vmid,
+						Version: 0,
+						Spec:    models.MicroVMSpec{},
+						Status: models.MicroVMStatus{
+							State: models.CreatedState,
+						},
+					},
+					nil,
+				)
+			},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			RegisterTestingT(t)
+
+			mockCtrl := gomock.NewController(t)
+			cm := mock.NewMockMicroVMCommandUseCases(mockCtrl)
+			qm := mock.NewMockMicroVMQueryUseCases(mockCtrl)
+
+			tc.expect(cm.EXPECT(), qm.EXPECT())
+
+			ctx := context.Background()
+			svr := grpc.NewServer(cm, qm)
+			resp, err := svr.CreateMicroVM(ctx, tc.createReq)
+
+			if tc.expectError {
+				Expect(err).To(HaveOccurred())
+			} else {
+				Expect(err).NotTo(HaveOccurred())
+				Expect(resp.Microvm.Id).To(Equal("mvm1"))
+				Expect(resp.Microvm.Namespace).To(Equal("default"))
+			}
+		})
+	}
+}
+
+func TestServer_DeleteMicroVM(t *testing.T) {
+	tt := []struct {
+		name        string
+		deleteReq   *mvm1.DeleteMicroVMRequest
+		expectError bool
+		expect      func(cm *mock.MockMicroVMCommandUseCasesMockRecorder, qm *mock.MockMicroVMQueryUseCasesMockRecorder)
+	}{
+		{
+			name:        "nil request should fail with error",
+			expectError: true,
+			expect:      func(cm *mock.MockMicroVMCommandUseCasesMockRecorder, qm *mock.MockMicroVMQueryUseCasesMockRecorder) {},
+		},
+		{
+			name:        "missing id should fail with error",
+			deleteReq:   &mvm1.DeleteMicroVMRequest{Id: "", Namespace: "default"},
+			expectError: true,
+			expect:      func(cm *mock.MockMicroVMCommandUseCasesMockRecorder, qm *mock.MockMicroVMQueryUseCasesMockRecorder) {},
+		},
+		{
+			name:        "missing namespace should fail with error",
+			deleteReq:   &mvm1.DeleteMicroVMRequest{Id: "mvm1", Namespace: ""},
+			expectError: true,
+			expect:      func(cm *mock.MockMicroVMCommandUseCasesMockRecorder, qm *mock.MockMicroVMQueryUseCasesMockRecorder) {},
+		},
+		{
+			name:        "error from usecase should fail with error",
+			deleteReq:   &mvm1.DeleteMicroVMRequest{Id: "mvm1", Namespace: "default"},
+			expectError: true,
+			expect: func(cm *mock.MockMicroVMCommandUseCasesMockRecorder, qm *mock.MockMicroVMQueryUseCasesMockRecorder) {
+				cm.DeleteMicroVM(
+					gomock.AssignableToTypeOf(context.Background()),
+					gomock.Not(gomock.Eq("")),
+					gomock.Not(gomock.Eq("")),
+				).Return(
+					errors.New("a random error occurred"),
+				)
+			},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			RegisterTestingT(t)
+
+			mockCtrl := gomock.NewController(t)
+			cm := mock.NewMockMicroVMCommandUseCases(mockCtrl)
+			qm := mock.NewMockMicroVMQueryUseCases(mockCtrl)
+
+			tc.expect(cm.EXPECT(), qm.EXPECT())
+
+			ctx := context.Background()
+			svr := grpc.NewServer(cm, qm)
+			_, err := svr.DeleteMicroVM(ctx, tc.deleteReq)
+
+			if tc.expectError {
+				Expect(err).To(HaveOccurred())
+			} else {
+				Expect(err).NotTo(HaveOccurred())
```

Comment:
This won't happen as there is no happy test case

Reply:
I should add a happy path....